src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -295,18 +284,13 @@ private void AddBody(InputParameter? bodyParameter, InputParameter? contentTypeR
             _protocolBodyParameter = bodyParameter.IsRequired
                 ? KnownParameters.RequestContent
                 : KnownParameters.RequestContentNullable;
-            _orderedParameters.Add(new ParameterChain(BuildParameter(bodyParameter), _protocolBodyParameter, _protocolBodyParameter));
 
-            if (contentTypeRequestParameter != null)
+            var convenienceBodyParameter = Parameter.FromInputParameter(bodyParameter, _typeFactory);
+            _orderedParameters.Add(new ParameterChain(convenienceBodyParameter, _protocolBodyParameter, _protocolBodyParameter));
+
+            if (contentTypeRequestParameter != null && Operation.RequestMediaTypes?.Count > 0)
             {
-                if (Operation.RequestMediaTypes?.Count > 1)
-                {
-                    AddContentTypeRequestParameter(contentTypeRequestParameter, Operation.RequestMediaTypes);
-                }
-                else
-                {
-                    AddParameter(contentTypeRequestParameter, typeof(ContentType));
-                }
+                AddContentTypeParameter(contentTypeRequestParameter, Operation.RequestMediaTypes);
```

Comment:
When contentTypeRequestParameter is not null, but it does not define RequestMediaTypes, will we not add this contentTypeRequestParameter? 
In that way, will we not set `content-type` or we will set default 'application/json' when create***Request?

Reply:
This is not possible in swagger (generated M4 should contain at least one media type). Is it possible in CADL?